cqrs/src/main/scala/it/pagopa/interop/commons/cqrs/service/ReadModelService.scala

```diff
@@ -0,0 +1,71 @@
+package it.pagopa.interop.commons.cqrs.service
+
+import cats.implicits._
+import it.pagopa.interop.commons.cqrs.errors.ReadModelErrors.ReadModelMissingDataField
+import it.pagopa.interop.commons.cqrs.model.ReadModelConfig
+import it.pagopa.interop.commons.utils.TypeConversions._
+import org.mongodb.scala.MongoClient.DEFAULT_CODEC_REGISTRY
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.connection.NettyStreamFactoryFactory
+import org.mongodb.scala.model.Projections
+import org.mongodb.scala.{ConnectionString, Document, MongoClient, MongoClientSettings, MongoDatabase}
+import spray.json._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+final class ReadModelService(dbConfig: ReadModelConfig) {
+
+  private val client: MongoClient = MongoClient(
+    MongoClientSettings
+      .builder()
+      .applyConnectionString(new ConnectionString(dbConfig.connectionString))
+      .codecRegistry(DEFAULT_CODEC_REGISTRY)
+      .streamFactoryFactory(NettyStreamFactoryFactory())
+      .build()
+  )
+
+  private val db: MongoDatabase = client.getDatabase(dbConfig.dbName)
+
+  def findOne[T: JsonReader](collectionName: String, filter: Bson)(implicit ec: ExecutionContext): Future[Option[T]] =
+    find[T](collectionName, filter, offset = 0, limit = 1).map(_.headOption)
+
+  def find[T: JsonReader](collectionName: String, filter: Bson, offset: Int, limit: Int)(implicit
+    ec: ExecutionContext
+  ): Future[Seq[T]] = find[T](collectionName, filter, Projections.include(), offset, limit)
+
+  def find[T: JsonReader](collectionName: String, filter: Bson, projection: Bson, offset: Int, limit: Int)(implicit
+    ec: ExecutionContext
+  ): Future[Seq[T]] =
+    for {
+      results <- db
+        .getCollection(collectionName)
+        .find(filter)
+        .projection(projection)
+        .skip(offset)
+        .limit(limit)
+        .toFuture()
+      model   <- results.traverse(extractData[T](_).toFuture)
+    } yield model
+
+  def aggregate[T: JsonReader](collectionName: String, pipeline: Seq[Bson])(implicit
+    ec: ExecutionContext
+  ): Future[Seq[T]] =
+    for {
+      results <- db
+        .getCollection(collectionName)
+        .aggregate(pipeline)
+        .toFuture()
+      model   <- results.traverse(extractData[T](_).toFuture)
```

Comment:
Both the traverse methods are using cats' `traverse` method over List, using `Future.traverse` is recommended